rfc/image-discoverability.md

```diff
@@ -0,0 +1,127 @@
+- a lookup table that maps the `image version uid` to the CSP-specific image references
+
+The lookup table is uploaded to S3 and is used to identify the images that belong to a given `image version uid`.
+Measurements are uploaded to S3 and can be looked up for each cloud service provider and `image version uid`.
+
+## Image lookup table
+
+The image lookup table is a JSON file that maps the `image version uid` to the CSP-specific image references. It uses the `image version uid` as file name.
+
+```
+s3://<BUCKET-NAME>/images/<IMAGE-VERSION-UID>.json
+```
+
+```json
+{
+  "aws": {
+    "us-east-1": "ami-123",
+    "us-west-2": "ami-456",
+    "eu-west-1": "ami-789"
+  },
+  "azure": {
+    "cvm": "azure-cvm-123",
+    "trustedlaunch": "azure-trusted-launch-123"
+  },
+  "gcp": "gcp-image-123",
+  "qemu": "https://cdn.edgeless.systems/.../qemu.raw"
+}
+```
+
+- For AWS, the image lookup table contains the AMI IDs for each region
+- For Azure, the image lookup table contains the image IDs for the CVM and Trusted Launch images
+- For GCP, the image lookup table contains the image ID
+- For QEMU, the image lookup table contains a URL to the QEMU image
+
+This document is not signed and can be extended in the future to include more image references (e.g. if an image is replicated to a new AWS region).
+The same document can be used to identify old images that are no longer used and can be deleted for cost optimization.
+
+## Image measurements
+
+This RFC is not about the image measurements themselves, but about how they are stored and looked up.
+The format of the image measurements is described in the [secure software distribution RFC](secure-software-distribution.md).
+
+The image measurements are stored in a folder structure in S3 that is organized by CSP and `image version uid`.
+
+```
+s3://<BUCKET-NAME>/measurements/<CSP>/<IMAGE-VERSION-UID>/measurements.yaml
+s3://<BUCKET-NAME>/measurements/<CSP>/<IMAGE-VERSION-UID>/measurements.yaml.sig
+```
+
+## CLI image discovery
+
+The CLI needs to be able to discover the image references for a given `image version uid`.
+By default, the CLI will prefill the `image` field of the `constellation-conf.yaml` when `constellation config generate <CSP>` is run with a hardcoded `image version uid` (e.g. `v2.2.0-cc0de5c68d41f31dd0b284d574f137e0b0ad106b`).
+The `image` field is independent of the CSP and is a used to discover the CSP-specific image reference as needed for the following operations:
+
+- `constellation create`
+- `constellation upgrade apply`
+
+The CLI can find a CSP- and region specific image reference by looking up the `image version uid` in the following order:
+
+- if a local file `<IMAGE-VERSION-UID>.json` exists, use the lookup table in that file
+- otherwise, load the image lookup table from a well known URL (e.g. `https://cdn.edgeless.systems/images/<IMAGE-VERSION-UID>.json`) and use the lookup table in that file
```

Comment:
Nit:
```suggestion
- otherwise, load the image lookup table from a well known URL (e.g. `https://cdn.confidential.cloud/images/<IMAGE-VERSION-UID>.json`) and use the lookup table in that file
```